File: branches/server/release_1.1/common/src/main/org/avis/io/messages/ConnRply.java
package org.avis.io.messages;

import java.util.Map;

import org.apache.mina.common.ByteBuffer;
import org.apache.mina.filter.codec.ProtocolCodecException;

import static org.avis.io.XdrCoding.getNameValues;
import static org.avis.io.XdrCoding.putNameValues;

public class ConnRply extends XidMessage
{
  public static final int ID = 50;

  /** Options requested by client that are supported. */
  public Map<String, Object> options;
  
  public ConnRply ()
  {
    // zip
  }
  
  public ConnRply (ConnRqst inReplyTo, Map<String, Object> options)
  {
    super (inReplyTo);
    
    this.options = options;
  }

  @Override
  public int typeId ()
  {
    return ID;
  }
  
  @Override
  public void encode (ByteBuffer out)
    throws ProtocolCodecException
  {
    super.encode (out);
    
    putNameValues (out, options);
  }
  
  @Override
  public void decode (ByteBuffer in)
    throws ProtocolCodecException
  {
    super.decode (in);
    
    options = getNameValues (in);
  }
}

File: mina_2.0_conversion/cpp/common/src/main/ConnRply.cpp
#include "ConnRply.h"

ConnRply::ConnRply()
{
}

ConnRply::ConnRply(ConnRqst &inReplyTo, ConnectionOptions theoptions)
: XidMessage(inReplyTo)
{
	options = theoptions;
}

ConnRply::~ConnRply()
{
}

void ConnRply::encode(XdrCoding *out)
{
	XidMessage::encode(out);

	out->putNameValues(&options);
}

void ConnRply::decode(XdrCoding *in)
{
	XidMessage::decode(in);

	in->getNameValues(&options);
}
